src/aof.c

```diff
@@ -833,6 +862,50 @@ int openNewIncrAofForAppend(void) {
     return C_ERR;
 }
 
+/* When we close gracefully the AOF file, we have the chance to persist the
+ * end replication offset of current INCR AOF. */
+void updateCurIncrAofEndOffset(void) {
+    if (server.aof_state != AOF_ON) return;
+    serverAssert(server.aof_manifest != NULL);
+
+    if (listLength(server.aof_manifest->incr_aof_list) == 0) return;
+    aofInfo *ai = listNodeValue(listLast(server.aof_manifest->incr_aof_list));
+    ai->end_offset = server.master_repl_offset;
```

Comment:
if we don't use repl, `master_repl_offset` should always be 0, should we skip this case?

Reply:
you may miss this change https://github.com/redis/redis/pull/11713/files#r1921898221

Reply:
yeah, i missed it.